sympy/vector/operators.py

```diff
@@ -247,6 +254,21 @@ def divergence(vect, coord_sys=None, doit=True):
         if doit:
             return (vx + vy + vz).doit()
         return vx + vy + vz
+    else:
+        if isinstance(vect, (Add, VectorAdd)):
+            return Add.fromiter(divergence(i) for i in vect.args)
+        elif isinstance(vect, (Mul, VectorMul)):
+            s = _split_mul_args_wrt_coordsys(vect)
+            scalar = [i for i in s if not isinstance(i, Vector) and len(_get_coord_systems(i)) > 0]
+            const = [i for i in s + [S(1)] if len(_get_coord_systems(i)) == 0][0]
+            vector = [i for i in s if isinstance(i, Vector)][0]
```

Comment:
As I wrote on gitter, you don't need all of this. A vector contains at most one vector symbol, the rest is a scalar. So just find the vector and apply the product rule.